### PR TITLE
+htc #17680 add `HttpEntity.contentLengthOption`

### PR DESCRIPTION
**Pull request by jrudolph**
/cc @mhamrah @briantopping @sirthias

### CODE REVIEW

**Comment by mhamrah** [comment]
LGTM (assuming you don't want to break backwards compatibility by just having contentLength be an Option[Long])

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala [hunk 1]
```diff
@@ -83,6 +95,10 @@ sealed trait HttpEntity extends jm.HttpEntity {
   /** Java API */
   def getDataBytes: stream.javadsl.Source[ByteString, _] = stream.javadsl.Source.adapt(dataBytes)
 
+  /** Java API */
+  def getContentLengthOption: japi.Option[JLong] =
+    japi.Option.fromScalaOption(contentLengthOption.asInstanceOf[Option[JLong]]) // Scala autoboxing
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: yeah that's really weird at times, +1

**Comment by ktoso** [comment]
LGTM

**Comment by jrudolph** [comment]
@mhamrah I just don't wanted to rename the existing `contentLength` members of Default and Strict which already have the perfect names and are the more important ones.

**Comment by drewhk** [comment]
LGTM,

needs a rebase

**Comment by jrudolph** [comment]
Rebased.

**Comment by ktoso** [comment]
Refs #17680, LGTM; Will merge after validation

**Comment by jrudolph** [comment]
TlsSpec

**Comment by drewhk** [comment]
Fixes #17680

**Pull request merged**